core/unix/module_elf.c

```diff
@@ -1528,7 +1535,7 @@ module_relocate_symbol(ELF_REL_TYPE *rel, os_privmod_data_t *pd, bool is_rela)
 
     res = module_lookup_symbol(sym, pd);
     LOG(GLOBAL, LOG_LOADER, 3, "symbol lookup for %s %p\n", name, res);
-    if (res == NULL && ELF_ST_BIND(sym->st_info) != STB_WEAK) {
+    if (res == NULL && ELF_ST_BIND(sym->st_info) != STB_WEAK && pd->soname != NULL) {
```

Comment:
We would still want a warning about undefined symbols, even if it isn't able to give a library name (I guess we don't have access to file_name here), so I would suggest entering this if() and just using `pd->soname == NULL ? "<empty soname>" : pd->soname` or something in the SYSLOG.